### DISCUSSION TITLE
Observable List of Devices

### DISCUSSION THREAD

**Discussion by SNewtown** [orchetect/swift-midi#216] [Q&A] [answered]
Hi Steffan,

I noticed in the MidiKit documentation that there is an observable list of devices.  My goal is to list (on the main screen of my iOS app) the connected devices and have this react to any newly connected/disconnected devices).  However, I can't seem to instantiate this correctly, can you provide some insight or point me to an example maybe?

I am on Midikit 0.8.8

Cheers,
Stephane

**Comment by orchetect** [comment, marked as answer]
I'd recommend updating to the latest version of the library. Check out the MIDIKitUI demo.

**Reply by SNewtown** [reply]
Thanks Steffan,

I can certainly give that a whirl, when I did update previously, I had quite a few breaks in my code, but I can try that again.  I agree with being on the latest, but is this mostly to be current or is 0.8.8 not able to expose the device list as an observable item?

Cheers,
Stephane

**Comment by SNewtown** [comment]
Updated to the latest version and ObservableDevices did the trick, works like a charm!  Thanks again Steffan, this midikit is something else, I must say, quite the time saver and super convenient!

**Reply by orchetect** [reply]
No problem. That's the whole reason I built the library so I'm glad it's doing its job (mostly). 😆
